app/src/main/java/org/simple/clinic/summary/PatientSummaryScreenController.kt

```diff
@@ -450,15 +453,46 @@ class PatientSummaryScreenController @Inject constructor(
   }
 
   private fun showUpdatePhoneDialogIfRequired(events: Observable<UiEvent>): Observable<UiChange> {
-    val patientUuidStream = events
-        .ofType<PatientSummaryScreenCreated>()
+    val screenCreations = events.ofType<PatientSummaryScreenCreated>()
+
+    val showForInvalidPhone = screenCreations
         .map { it.patientUuid }
+        .switchMap { patientUuid ->
+          hasInvalidPhone(patientUuid)
+              .take(1)
+              .filter { invalid -> invalid }
+              .map { { ui: Ui -> ui.showUpdatePhoneDialog(patientUuid) } }
+        }
 
-    return patientUuidStream
-        .switchMap { patientUuid -> Observables.zip(phoneNumber(patientUuid), lastCancelledAppointment(patientUuid)) }
-        .filter { (number, appointment) -> appointment.updatedAt > number.updatedAt }
-        .take(1)
-        .map { (number) -> { ui: Ui -> ui.showUpdatePhoneDialog(number.patientUuid) } }
+    val showForMissingPhone = screenCreations
+        .filter { it.caller == SEARCH }
```

Comment:
Agreed. Any suggestion for the name?

Reply:
`isLatestAppointmentCancelledWithInvalidPhone`? Otherwise, I'd move that filter out into where it's being used so that this filter happens where we are checking if the phone number is incorrect or not.

Reply:
I'll just move the filter to avoid this long name :p

Reply:
Actually I can't escape this. I'm just going to use your name suggestion.